tests/taskgraphs/test_codec.py

```diff
@@ -99,9 +99,74 @@ def test_unescaping(self):
             ),
         )
 
-        for name, inp, expected in cases:
-            with self.subTest(name):
-                decoded_input = json.loads(inp)
+        for name, json_text, native_value in cases:
+            json_data = json.loads(json_text)
+            with self.subTest(f"unescaping {name}"):
                 unesc = _codec.Unescaper()
-                actual = unesc.visit(decoded_input)
-                self.assertEqual(expected, actual)
+                actual = unesc.visit(json_data)
+                self.assertEqual(native_value, actual)
+
+            with self.subTest(f"escaping {name}"):
+                esc = _codec.Escaper()
+                actual = esc.visit(native_value)
+                self.assertEqual(json_data, actual)
+
+    def test_json_encodable(self):
+        data = _codec.BinaryResult("bytes", b"now")
+        esc = _codec.Escaper()
+        in_data = ["one", data, 2]
+        expected = [
+            "one",
+            {"__tdbudf__": "immediate", "format": "bytes", "base64_data": "bm93"},
+            2,
+        ]
+        self.assertEqual(expected, esc.visit(in_data))
+
+
+class BinaryResultTest(unittest.TestCase):
+    def test_binary_result_of(self):
+        cases = (
+            (b"possession", _codec.BinaryResult("bytes", b"possession")),
+            (
+                frozenset(),
+                _codec.BinaryResult(
+                    "python_pickle",
+                    b"\x80\x04\x95\x04\x00\x00\x00\x00\x00\x00\x00(\x91\x94.",
```

Comment:
can you add a comment how this hex string was created, so a less-Python-expert-than-Paul person could reproduce this in the future, e.g. if they ever try to add similar test cases?

Reply:
i am guessing:

```
>>> import cloudpickle
>>> cloudpickle.dumps(b"possession")
b'\x80\x04\x95\x0e\x00\x00\x00\x00\x00\x00\x00C\npossession\x94.'
```

but that's not quite right as I'm not reproducing your line 134

also, please let a comment here regarding what Python version was used here